bot/database/__init__.py

```diff
@@ -0,0 +1,267 @@
+import typing as t
+from importlib import import_module
+
+import asyncpg
+from loguru import logger
+
+if t.TYPE_CHECKING:
+    from bot.core.bot import Bot
```

Comment:
Uhmmm, What's this?

Reply:
This is a way to prevent circular imports and yet still import `Bot`

The thing that in `bot.core.bot` we're importing `bot.database` which means you can't import `bot.core.bot` from `bot.database` that will just lead to an import loop where bot will import db, which will import bot, which will import db, etc.

Since the only reason `Bot` is imported here is because of the type hints. `typing` module has this covered by doing `if tYPING.TYPE_CHECKING` and putting these imports under that.